tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_inlining.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# GTC Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from gt4py.backend.gtc_backend.defir_to_gtir import DefIRToGTIR
+from gt4py.definitions import BuildOptions
+from gt4py.frontend.gtscript_frontend import GTScriptFrontend
+from gt4py.gtscript import PARALLEL, Field, computation, interval
+from gtc.gtir_to_oir import GTIRToOIR
+from gtc.oir import BinaryOp, FieldAccess
+from gtc.passes.gtir_pipeline import GtirPipeline
+from gtc.passes.oir_optimizations.inlining import MaskInlining
+
+
+def test_mask_inlining():
+    def stencil_def(
+        extm: Field[float],
+        a4_1: Field[float],
+        a4_2: Field[float],
+    ):
+        with computation(PARALLEL), interval(...):
+            if extm != 0.0 and (extm[0, 0, -1] != 0.0 or extm[0, 0, 1] != 0.0):
+                a4_2 = a4_1
+            else:
+                a4_2 = 6.0 * a4_1 - 3.0 * a4_2
```

Comment:
I believe OIR tests should not depend on the gtscript frontend. Most of the others build up an OIR directly. Could this be changed to that pattern?

Reply:
I borrowed this approach from the `test_dace` [code](https://github.com/GridTools/gt4py/blob/25aab82cc076a64eb2ecfed9e146d620628464e0/tests/test_unittest/test_gtc/test_dace.py#L51).

Reply:
It's probably the only test you should not take as inspiration for test patterns of oir passes. It's a very special case (and probably shouldn't be in unittest) as it's testing a round trip, i.e. for all possible stencil inputs you can validate correct results.
As @jdahm said, all the other tests should start at oir, apply the transformation and validate the resulting oir to decouple it from all the rest of GT4Py.